### ISSUE TITLE
Switch to Lexical Binding?

### ISSUE THREAD

**Issue by xuchunyang** [eschulte/emacs-web-server#14]
This package requires Emacs 24.3 and Emacs 24.1 supports lexical binding, so this package can use lexical binding. I assuming lexical binding is preferred nowadays.

I tried the following change, recompile, restart Emacs, and web-server seems works as usual, maybe we don't need other change at all.
 
```diff
-;;; web-server.el --- Emacs Web Server
+;;; web-server.el --- Emacs Web Server  -*- lexical-binding: t; -*-
```

**Comment by eschulte** [comment]
Can you suggest any concrete improvements you anticipate from enabling lexical binding?

**Comment by eschulte** [comment]
Okay, I've just addressed this issue in f58ecc6.  Thanks for the report!

**Comment by xuchunyang** [comment]
> Lexical binding opens up many more opportunities for optimization, so programs using it are likely to run faster in future Emacs versions
> https://www.gnu.org/software/emacs/manual/html_node/elisp/Lexical-Binding.html


Faster code execution. Better byte compilation report.
